File: smart_travel_FE/lib/data/models/banner/banner_create_request.dart
class BannerCreateRequest {
  final String title;
  final String imageUrl;
  final String? linkUrl;
  final String? description;

  BannerCreateRequest({
    required this.title,
    required this.imageUrl,
    this.linkUrl,
    this.description,
  });

  // Chuyển đổi đối tượng sang Map để encode sang JSON
  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'imageUrl': imageUrl,
      'linkUrl': linkUrl,
      'description': description,
    };
  }
}

File: smart_travel_BE/src/main/java/com/example/smart_travel_BE/dto/banner/request/BannerCreateRequest.java
package com.example.smart_travel_BE.dto.banner.request;

import jakarta.validation.constraints.NotBlank;
import lombok.*;
import lombok.experimental.FieldDefaults;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class BannerCreateRequest {
    @NotBlank(message = "BANNER_NAME_BLANK")
    String title;

    @NotBlank(message = "IMAGE_URL_BLANK")
    String imageUrl;

    String linkUrl;
    String description;
}